tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,360 @@
+from collections import defaultdict
+import matplotlib.cm as cm
+import matplotlib.colors as clr
+import plotly.graph_objects as go
+from IPython.display import display
+import matplotlib as mpl
+import ipywidgets as widgets
+from contextlib import suppress
+from traitlets import TraitError
+
+
+def transition_colors(name="jet", iterations=20):
+    """
+    Function to create colorscale for convergence plots, returns a list of colors.
+
+    Parameters
+    ----------
+    name : string
+        Name of the colorscale. Defaults to "jet".
+    iterations : int
+        Number of iterations. Defaults to 20.
+    Returns
+    -------
+    colors: list
+    """
+    cmap = mpl.cm.get_cmap(name, iterations)
+    colors = []
+    for i in range(cmap.N):
+        rgb = cmap(i)[:3]
+        colors.append(mpl.colors.rgb2hex(rgb))
+    return colors
+
+
+class ConvergencePlots(object):
+    """
+    Class to create and update convergence plots for visualizing convergence of the
+    simulation.
+
+    Parameters
+    ----------
+    iteration : int
+        iteration number
+    """
+
+    def __init__(self, iterations, **kwargs):
+        self.iterable_data = {}
+        self.value_data = defaultdict(list)
+        self.iterations = iterations
+        self.current_iteration = 1
+        self.luminosities = ["Emitted", "Absorbed", "Requested"]
+        self.plasma_plot = None
+        self.luminosity_plot = None
+
+        if "colorscale" in kwargs:
+            self.colorscale = transition_colors(
+                kwargs["colorscale"], iterations=self.iterations
+            )
+        else:
+            self.colorscale = transition_colors(iterations=self.iterations)
+
+        if "plasma_plot_config" in kwargs:
+            if kwargs["plasma_plot_config"] != {}:
```

Comment:
Better to `and` the both. Also `if x != {}` is equivalent to `if x` - we're checking truthy/falsy values